packages/tracing/src/integrations/postgres.ts

```diff
@@ -0,0 +1,81 @@
+import { Hub } from '@sentry/hub';
+import { EventProcessor, Integration } from '@sentry/types';
+import { fill, logger } from '@sentry/utils';
+
+interface PgClient {
+  prototype: {
+    query: () => void | Promise<unknown>;
+  };
+}
+
+interface PostgresOptions {
+  client?: PgClient;
+}
+
+/** Tracing integration for node-postgres package */
+export class Postgres implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'Postgres';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = Postgres.id;
+
+  private _client?: PgClient;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(options: PostgresOptions = {}) {
+    this._client = options.client;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    if (!this._client) {
+      logger.error('PostgresIntegration is missing a Postgres.Client constructor');
+      return;
+    }
+
+    /**
+     * function (query, callback) => void
+     * function (query, params, callback) => void
+     * function (query) => Promise
+     * function (query, params) => Promise
+     */
+    fill(this._client.prototype, 'query', function(orig: () => void | Promise<unknown>) {
```

Comment:
I assume the entries in the docstring are the various forms the original function can take? Then why type `orig` as taking no arguments?

Reply:
Because `orig` is variadic, and typing it out would be just pain in the ass, which brings 0 value to the table. It's not user facing and we have to base the code on the original implementation anyway.